test/components/personkort/PersonkortHeaderTest.tsx

```diff
@@ -37,46 +37,81 @@ describe("PersonkortHeader", () => {
 
   it("viser 'Egenansatt' når isEgenansatt er true fra API", async () => {
     stubEgenansattApi(apiMockScope, true);
-    stubDiskresjonskodeApi(apiMockScope);
     renderPersonkortHeader();
 
     expect(await screen.findByText("Egenansatt")).to.exist;
   });
 
   it("viser ikke 'Egenansatt' når isEgenansatt er false fra API", async () => {
     stubEgenansattApi(apiMockScope, false);
-    stubDiskresjonskodeApi(apiMockScope, "6");
     renderPersonkortHeader();
-    await screen.findByText("Kode 6");
 
     expect(screen.queryByText("Egenansatt")).not.to.exist;
   });
 
   it("viser 'Kode 6' når diskresjonskode er 6 fra API", async () => {
-    stubEgenansattApi(apiMockScope, true);
     stubDiskresjonskodeApi(apiMockScope, "6");
     renderPersonkortHeader();
 
     expect(await screen.findByText("Kode 6")).to.exist;
   });
 
   it("viser 'Kode 7' når diskresjonskode er 7 fra API", async () => {
-    stubEgenansattApi(apiMockScope, true);
     stubDiskresjonskodeApi(apiMockScope, "7");
     renderPersonkortHeader();
 
     expect(await screen.findByText("Kode 7")).to.exist;
   });
 
   it("viser ingen diskresjonskode når diskresjonskode er tom fra API", async () => {
-    stubEgenansattApi(apiMockScope, true);
     stubDiskresjonskodeApi(apiMockScope);
     renderPersonkortHeader();
-    await screen.findByText("Egenansatt");
 
     expect(screen.queryByText("Kode")).not.to.exist;
   });
 
+  it("viser ikke tegnspråktolk eller talespråktolk når tilrettelagtKommunikasjon er null fra API", async () => {
+    const tilrettelagtKommunikasjon = {
+      talesprakTolk: null,
+      tegnsprakTolk: null,
+    };
+    stubPersoninfoApi(apiMockScope, "", tilrettelagtKommunikasjon);
+    renderPersonkortHeader();
+
+    expect(screen.queryByText("Talespråktolk")).not.to.exist;
+    expect(screen.queryByText("Tegnspråktolk")).not.to.exist;
+  });
+
+  it("viser talespråktolk, men ikke tegnspråktolk", async () => {
+    const tilrettelagtKommunikasjon = {
+      talesprakTolk: {
+        value: "NO",
+      },
+      tegnsprakTolk: null,
+    };
+    stubPersoninfoApi(apiMockScope, "", tilrettelagtKommunikasjon);
+    renderPersonkortHeader();
+
+    await waitFor(() => expect(screen.queryByText("Talespråktolk")).to.exist);
+    expect(screen.queryByText("Tegnspråktolk")).to.not.exist;
+  });
```

Comment:
Av en eller annen grunn fikk vi ikke til å awaite `stubPersoninfoApi`, så den eneste måten å enforce at det var riktig data i komponenten var en slik `waitfor` (se https://testing-library.com/docs/dom-testing-library/api-async/#waitfor) 🤔 Noen som vet hvorfor det går an å ta `await screen.findByText` i de andre testene, men ikke her?

Reply:
Fungerer med `expect(await screen.findByText("Talespråktolk")).to.exist;` lokalt her 💭